src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type != Bulldozer.class && type != Excavator.class && type != Truck.class) {
+            return new ArrayList<>();
+        }
+        MachineProducer<? extends Machine> machineProducer = new TruckProducer();
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+        }
+        if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
+
+        }
```

Comment:
lets create class-level fields with producers
don't create a new instance each time this method is invoked
```suggestion
        List<? extends Machine> machineList = new ArrayList<>();
        if (type == Bulldozer.class) {
            machineList = bulldozerProducer.get();
        }
        if (type == Excavator.class) {
            machineList = excavatorProducer.get();
        }
        if (type == Truck.class) {
            machineList = truckProducer,get();
        }
```